src/transitions/Fade.js

```diff
@@ -1,53 +1,60 @@
 // @flow weak
 
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Element, PureComponent } from 'react';
 import Transition from '../internal/Transition';
 import { duration } from '../styles/transitions';
 import customPropTypes from '../utils/customPropTypes';
 
-export default class Fade extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    /**
-     * If `true`, the component will transition in.
-     */
-    in: PropTypes.bool,
-    /**
-     * Duration of the animation when the element is entering.
-     */
-    enterTransitionDuration: PropTypes.number, // eslint-disable-line react/sort-prop-types
-    /**
-     * Duration of the animation when the element is exiting.
-     */
-    leaveTransitionDuration: PropTypes.number,
-    /**
-     * Callback fired before the component enters.
-     */
-    onEnter: PropTypes.func,
-    /**
-     * Callback fired when the component is entering.
-     */
-    onEntering: PropTypes.func,
-    /**
-     * Callback fired when the component has entered.
-     */
-    onEntered: PropTypes.func, // eslint-disable-line react/sort-prop-types
-    /**
-     * Callback fired before the component exits.
-     */
-    onExit: PropTypes.func,
-    /**
-     * Callback fired when the component is exiting.
-     */
-    onExiting: PropTypes.func,
-    /**
-     * Callback fired when the component has exited.
-     */
-    onExited: PropTypes.func, // eslint-disable-line react/sort-prop-types
-  };
+type DefaultProps = {
+  in: boolean,
+  enterTransitionDuration: number,
+  leaveTransitionDuration: number,
+}
+
+type Props = DefaultProps & {
+  children?: Element<*>,
+  /**
+   * If `true`, the component will transition in.
+   */
+  in?: boolean,
+  /**
+   * Duration of the animation when the element is entering.
+   */
+  enterTransitionDuration?: number, // eslint-disable-line react/sort-prop-types
+  /**
+   * Duration of the animation when the element is exiting.
+   */
+  leaveTransitionDuration?: number,
+  /**
+   * Callback fired before the component enters.
+   */
+  onEnter?: Function,
+  /**
+   * Callback fired when the component is entering.
+   */
+  onEntering?: Function,
+  /**
+   * Callback fired when the component has entered.
+   */
+  onEntered?: Function, // eslint-disable-line react/sort-prop-types
+  /**
+   * Callback fired before the component exits.
+   */
+  onExit?: Function,
+  /**
+   * Callback fired when the component is exiting.
+   */
+  onExiting?: Function,
+  /**
+   * Callback fired when the component has exited.
+   */
+  onExited?: Function, // eslint-disable-line react/sort-prop-types
+};
+
+export default class Fade extends PureComponent<DefaultProps, Props, void> {
```

Comment:
As a rule of thumb, we should not use any pure logic with React elements. They are created at each render, making strict comparison always negative, hence, wasted CPU cycles.